proposals/0245-sharing-wifi-ssid-and-password.md

```diff
@@ -18,627 +18,1205 @@ If WiFi credentials are automatically generated by the IVI system, the password
+
+### Manual WiFi connection
+
+Mobile device is still can be connected to the Vehicle manually (or vice versa). In this case, if an application is connected via other transports, it will receive TransportEventUpdate and applications will be able to connect via WiFi, so the feature is not requred after this point. Core should not process `OnNetworkConfiguration` and `JoinNetwork` for the application, which is already connected via WiFi.
+
+However, if there are no applications connected via WiFi, Core knows nothing of the established connection, so the first feature enable application will still be able to use the feature.
+
+![WiFi AP created manually. Disconnected](../assets/proposals/0245-sharing-wifi-ssid-and-password/WiFi_AP_created_manually_Disconnected.png)
+
+![Negative cases](../assets/proposals/0245-sharing-wifi-ssid-and-password/negative_case.png)
+
+## Potential Downsides
+
+SDL is not able to check that 2 different transport belongs to the same device.
+On device connected by 2 different devices will be 2 devices for SDL. 
+
+## Alternative solutions 
+
+###  Minor alternatives : 
+ 1. Use OnSystemRequest instead of new notification.
+
+Proposition discussed and rejected during SDLC discussion  https://github.com/smartdevicelink/sdl_evolution/issues/799
+
+
+2. Do not postpone sending `OnNetworkConfiguration` to HMI if secure service was not started, but extend `OnServiceStatusUpdate` with `encrypted` flag. 
+ 
+ - This rejected because `OnServiceStatusUpdate` is specific for application, but for Wifi functionality sharing HMI does not know particular application for sending `JoinNetwork`. 
+
+
+3. Leave establishing secondary transport on app activation. 
+   Consider following drawbacks:
+   - Application will receive redundant JoinNetwork
+   - HMI will show redundant confusing user consent
+   - Application will receive redundant JoinNetwork on app on second device Register
+     applicable if Mobile is AP, and multiple devices (sharing is allowed)
+
+## Minimal approach solution
+
+### Proposed solution
+
+Current proposal introduces additional APIs for sharing Wifi credentials between the vehicle and devices.
+
+This solution does not put on SDL any logic to determine access point host.
+This decision should be controlled by HMI.
+
+SDL does not have direct access to Wifi hardware, so SDL is not able to check the current Wifi state.
+Such an approach gives OEM the ability to have own rules and priorities for Wifi interaction.
+
+This approach can be easily integrated into some already existing solution because there is no additional logic on SDL, and OEM may just adopt an existing solution for using new APIs.

```

Comment:
```suggestion
This approach can be easily integrated into some already existing solution because there is no additional logic on SDL Core, and an OEM may just adopt an existing solution for using new APIs.
```